### PR TITLE
Apply landmask to fields for gpnorm() and random()

### LINKED ISSUE #114: gpnorm with mask
The current gpnorm is only taking account of the mask when counting the # of grid points. We need to constrain the gpnorm stats to the unmasked values.

### LINKED ISSUE #231: mask land in Fields::random()
Currently, randomly generated fields from `Fields::random()` have random values everywhere. We should be masking out the land and setting those to 0. This will be needed so that the `GeometryIterator` test being added in #225 passes if the `GeometryIterator` skips over land (which we'll want to do, since there is no point doing the LETKF on land for the ocean analysis)

related somewhat to #114

### PR DESCRIPTION
**Pull request by travissluka**
* applies the land mask for `Increment::random()`, land locations are set to zero for the random increment. This is needed so that the `GeometryIterator` test will pass if the soca iterator skips over land points. This currently does not change the answers of any of our tests
* instead of doing the same thing for `ones()`, I just removed that method because it was not being used and is not required by the oops interface. (Free bonus code coverage points!!)
* the same masking is applied to `gpnorm()`, min/max/mean values now exclude land points (e.g. look at the minimum salinity, which is no longer 0.0). Test reference answers are updated because of this.

closes #114 
closes #231

### CODE REVIEW

**Review by guillaumevernieres — APPROVED** [review]
Approving but also noting that we lost our edge on the development of 1's Id and nothing classes ... sad.

File: src/soca/Fields/soca_fields_mod.F90 [hunk 1]
```diff
@@ -130,8 +130,8 @@ end subroutine create_copy
 ! ------------------------------------------------------------------------------
 
 subroutine soca_field_alloc(self, geom)
```
Comment (frolovsa) [inline src/soca/Fields/soca_fields_mod.F90:132]: doesn't look like input geom is used in this subroutine

**Review by frolovsa — COMMENTED** [review]
Looks good.

File: src/soca/Fields/soca_fields_mod.F90 (hunk 1 above)
Comment (guillaumevernieres) [reply src/soca/Fields/soca_fields_mod.F90:132]: Welcome @frolovsa !
Not sure what you mean? see L141-142, 154, 157

Comment (frolovsa) [reply src/soca/Fields/soca_fields_mod.F90:132]: @guillaumevernieres . Ok. i see now. Sorry about that. Using this new environment is like diving in a muddy river. You remember how to breath underwater but you can only see 3 inches in front of your face.  Thank you guys for your patience.

Comment (guillaumevernieres) [reply src/soca/Fields/soca_fields_mod.F90:132]: No worries @frolovsa , thanks for looking at the code. It's not bug-free for sure, so don't be shy and ask/post if something looks fishy.

**Pull request merged**